core/matching/request_template_store.go

```diff
@@ -99,18 +99,26 @@ func (this *RequestTemplateStore) Wipe() {
 /**
 Check keys and corresponding values in template headers are also present in request headers
 */
-func headerMatch(tmplHeaders, reqHeaders map[string][]string) bool {
+func headerMatch(templateHeaders, requestHeaders map[string][]string) bool {
 
-	for headerName, headerVal := range tmplHeaders {
-		// TODO: case insensitive lookup
-		// TODO: is order of values in slice really important?
-
-		reqHeaderVal, ok := reqHeaders[headerName]
-		if ok && reflect.DeepEqual(headerVal, reqHeaderVal) {
-			continue
-		} else {
+	for templateHeaderKey, templateHeaderValues := range templateHeaders {
+		requestTemplateValues, ok := requestHeaders[templateHeaderKey]
+		if !ok {
 			return false
 		}
+
+		for _, templateHeaderValue := range templateHeaderValues {
+			found := 0
+			for _, requestHeaderValue := range requestTemplateValues {
+				if glob.Glob(templateHeaderValue, requestHeaderValue) {
+					found++
```

Comment:
Good question.

I tried that and some tests failed which then made me remember why it is done like this.

In Hoverfly, the value of a header is an array of strings. There for, your template could potentially have multiple values to match on. Short-cutting the return when stop this functionality from working as you would return on your first match.

Your template:
```
Content-Type: [text/plain, application/json]
```

Unmatching request:
```
Content-Type: [text/plain]
```